### PR TITLE
Make table of span offsets optional in _repr_html

### PR DESCRIPTION
**Pull request by frreiss**
This PR adds a flag to turn off the table of detailed spans in the HTML (i.e. Jupyter notebook) representation of `TokenSpanArray` and `CharSpanArray`.  Removing the table is useful when the user is only interested in seeing the marked-up target text. The flag is called `repr_html_show_offsets` and is a separate property for each object.

The default value of `True` gives the original `_repr_html()` behavior:
repr_html_before

Setting the flag to `False` removes the table of span offsets:
repr_html_after

I have also updated the regression tests accordingly and eliminated some code duplication between `CharSpanArray` and `TokenSpanArray`.

### CODE REVIEW

File: text_extensions_for_pandas/array/token_span.py [hunk 1]
```diff
@@ -764,12 +764,6 @@ def as_frame(self) -> pd.DataFrame:
     ##########################################
     # Keep private and protected methods here.
 
-    def _repr_html_(self) -> str:
```
Comment (ZachEichen) [inline text_extensions_for_pandas/array/token_span.py:766]: why is this being removed?

Comment (ZachEichen) [reply text_extensions_for_pandas/array/token_span.py:766]: other than this it looks good to me!

Comment (frreiss) [reply text_extensions_for_pandas/array/token_span.py:766]: That function overrides a function by the same name, with the same function body, in the parent class `CharSpanArray`. I discovered the duplicate functions when I modified `CharSpanArray._repr_html()`. I think the duplicate functions are a relic of the time before `TokenSpanArray` was a subclass of `CharSpanArray`.

**Comment by frreiss** [comment]
Build passes now after merging #42. Merging this PR.

**Pull request merged**